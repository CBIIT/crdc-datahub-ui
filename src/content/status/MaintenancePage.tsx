import { FC, memo } from "react";
import { Box, Stack, Typography, styled } from "@mui/material";
import MaintenanceIcon from "../../assets/icons/maintenance.svg?react";
import usePageTitle from "../../hooks/usePageTitle";

const StyledContainer = styled(Stack)({
  background: "#2F4F70",
  minHeight: "100vh",
  paddingTop: "89px",
  paddingBottom: "140px",
});

const StyledContent = styled(Box)({
  textAlign: "center",
  margin: "0 auto",
  maxWidth: "795px",
});

const StyledPageTitle = styled(Typography)({
  color: "#31F7C3",
  fontWeight: 800,
  fontFamily: "Nunito Sans",
  fontSize: "45px",
});

const StyledBodyText = styled(Typography)({
  color: "#D2E9EE",
  marginTop: "10px",
  fontSize: "18px",
  fontFamily: "Inter",
  lineHeight: "26px",
});

/**
 * Provides a maintenance page for the application.
 *
 * If passed `state.data.shouldBlock` is true, it will block any navigation away.
 *
 * @returns The maintenance page component.
 */
const MaintenancePage: FC = () => {
  usePageTitle("Website Maintenance");

  return (
    <StyledContainer>
      <StyledContent>
        <MaintenanceIcon />
        <StyledPageTitle>Website Maintenance</StyledPageTitle>
        <StyledBodyText>
          The CRDC Submission Portal site is currently undergoing scheduled maintenance. Please
          check back soon. We appreciate your patience.
        </StyledBodyText>
      </StyledContent>
    </StyledContainer>
  );
};

export default memo<FC>(MaintenancePage);
