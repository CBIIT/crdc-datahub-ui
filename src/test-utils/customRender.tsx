import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import theme from "../theme/testTheme";

const AllTheProviders = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
