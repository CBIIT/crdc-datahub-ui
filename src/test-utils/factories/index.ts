export * from "./ApprovedStudyFactory";
export * from "./BatchFactory";
export * from "./SubmissionStatisticFactory";
export * from "./UserFactory";
