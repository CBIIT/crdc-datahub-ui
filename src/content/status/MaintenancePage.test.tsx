import { axe } from "vitest-axe";
import { render } from "../../test-utils";
import MaintenancePage from "./MaintenancePage";

const mockUsePageTitle = vi.fn();
vi.mock("../../hooks/usePageTitle", () => ({
  __esModule: true,
  default: (...p) => mockUsePageTitle(...p),
}));

describe("Accessibility", () => {
  it("should not have any accessibility violations", async () => {
    const { container } = render(<MaintenancePage />);

    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Basic Functionality", () => {
  it("should render without crashing", () => {
    expect(() => render(<MaintenancePage />)).not.toThrow();
  });
});

describe("Implementation Requirements", () => {
  it("should set the page title correctly", () => {
    render(<MaintenancePage />);
    expect(mockUsePageTitle).toHaveBeenCalledWith("Website Maintenance");
  });

  it("should display a user-friendly message", () => {
    const { getByText } = render(<MaintenancePage />);

    expect(getByText("Website Maintenance")).toBeInTheDocument();
    expect(
      getByText(
        "The CRDC Submission Portal site is currently undergoing scheduled maintenance. Please check back soon. We appreciate your patience."
      )
    ).toBeInTheDocument();
  });
});

describe("Snapshots", () => {
  it("should match the snapshot", () => {
    const { container } = render(<MaintenancePage />);

    expect(container).toMatchSnapshot();
  });
});
