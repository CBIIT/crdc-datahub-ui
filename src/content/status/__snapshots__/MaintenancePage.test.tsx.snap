// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should match the snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-pifray-MuiStack-root"
  >
    <div
      class="MuiBox-root css-1ao5c25"
    >
      <svg>
        maintenance.svg
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 css-moeelo-MuiTypography-root"
      >
        Website Maintenance
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-wd1mvi-MuiTypography-root"
      >
        The CRDC Submission Portal site is currently undergoing scheduled maintenance. Please check back soon. We appreciate your patience.
      </p>
    </div>
  </div>
</div>
`;
